urla134d66615_fikr4n_itl_java_tgz-pJ8-example_ExampleJ8/src/org/arabeyes/itl/prayer/PrayerTimeCalc.java

```diff
@@ -0,0 +1,720 @@
+		
+		/* Final Step: Fill the Prayer array by doing decimal degree to
+		 * Prayer structure conversion*/
+		if (type == InternalTimeType.IMSAK /* || type == TimeType.NEXT_FAJR */) {
+			base6hm(tempPrayer[0], method, pt.fajr(), type);
+		} else {
+			PrayerTime[] pArray = pt.getPrayerTime();
+			InternalTimeType[] timeArray = { InternalTimeType.FAJR, InternalTimeType.SUNRISE,
+					InternalTimeType.ZUHR, InternalTimeType.ASR, InternalTimeType.MAGHRIB,
+					InternalTimeType.ISHA };
+			
+			for (i = 0; i < 6; i++) {
+				base6hm(tempPrayer[i], method, pArray[i], timeArray[i]);
+			}
+		}
+		
+	}
+
+	private void base6hm(double bs, Method method, PrayerTime pt, InternalTimeType type) {
+		double min, sec;
+		
+		if (bs == 99) {
+			pt.setHour(99);
+			pt.setMinute(99);
+			pt.setSecond(0);
+			return;
+		}
+		
+		/* Add offsets */
+		if (method.getOffset()) {
+			if (type == InternalTimeType.IMSAK /* || type == TimeType.NEXT_FAJR */)
+				bs += (method.getFajrOffset() / 60.0);
+			else
+				bs += (method.getOffset(type) / 60.0);
+		}
+		
+		/* Fix after minus offsets before midnight */
+		if (bs < 0) {
+			while (bs < 0)
+				bs = 24 + bs;
+		}
+		
+		min = (bs - Math.floor(bs)) * 60;
+		sec = (min - Math.floor(min)) * 60;
+		
+		/* Add rounding minutes */
+		if (method.getRound() == Rounding.NORMAL) {
+			if (sec >= Utils.DEFAULT_ROUND_SEC)
+				bs += 1 / 60.0;
+			/* compute again */
+			min = (bs - Math.floor(bs)) * 60;
+			sec = 0;
+			
+		} else if (method.getRound() == Rounding.SPECIAL || method.getRound() == Rounding.AGRESSIVE) {
+			if (type == InternalTimeType.FAJR || type == InternalTimeType.ZUHR
+					|| type == InternalTimeType.ASR || type == InternalTimeType.MAGHRIB
+					|| type == InternalTimeType.ISHA /* || type == TimeType.NEXT_FAJR */) {
+				if (method.getRound() == Rounding.SPECIAL) {
+					if (sec >= Utils.DEFAULT_ROUND_SEC) {
+						bs += 1 / 60.0;
+						min = (bs - Math.floor(bs)) * 60;
+					}
+				} else if (method.getRound() == Rounding.AGRESSIVE) {
+					if (sec >= Utils.AGGRESSIVE_ROUND_SEC) {
+						bs += 1 / 60.0;
+						min = (bs - Math.floor(bs)) * 60;
+					}
+				}
+				sec = 0;
+			} else {
+				//case Prayer.SHUROOQ:
+				//case Prayer.IMSAAK:
+				sec = 0;
+			}
+		}
+		
+		/* Add daylight saving time and fix after midnight times */
+		bs += loc.getDst();
+		if (bs >= 24) {
+			bs = Math.IEEEremainder(bs, 24);
+		}
+		
+		pt.setHour((int) bs);
+		pt.setMinute((int) min);
+		pt.setSecond((int) sec);
+		
+	}
+
+	private static double getFajIsh(double Lat, double dec, double Ang) {
+		
+		double part1 = Math.cos(Utils.degToRad(Lat)) * Math.cos(dec);
+		double part2 = -Math.sin(Utils.degToRad(Ang))
+		- Math.sin(Utils.degToRad(Lat)) * Math.sin(dec);
+		
+		double part3 = part2 / part1;
+		if (part3 <= Utils.INVALID_TRIGGER) {
+			return 99;
+		}
+		
+		return Utils.DEG_TO_10_BASE * Utils.radToDeg(Math.acos(part3));
+		
+	}
+	
+	private static double getShoMag(Location loc, Astro astro, InternalTimeType type) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 45 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 98% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 599-644 into a separate method.